src/routes/private.ts

```diff
@@ -33,6 +34,15 @@ export default async function (fastify: FastifyInstance): Promise<void> {
 
     const body = { ...rest, referralId, referralOrigin };
     const operationResult = await addNewUser(con, body, req.log);
+
+    if (operationResult.status === 'ok') {
```

Comment:
Ideally this new insert should be in one transaction with the adding the user. Since `addNewUser` is now a complex process that is not capable of being in a transaction this process may leave us with zombie users without a digest because we don't handle the error. Another approach would be a database trigger on insert. It is more failsafe. @capJavert wdyt? Sorry for being annoying

Reply:
@idoshamun I did not do it in a transaction because the reference integrity was failing due to user not existing yet in the DB even though it does exist in transaction.

Trigger is a good idea but sounds like a hassle if we wanna add any kind of logic on when we subscribe users to digest.

WDYT?

Reply:
With worker we did have automatic retry which was good.

Reply:
When talking about the timezone I am not sure we actually need the timezone on digest then. If we just update that based on `user` table `timezone` column what is the point of it being in `user_personal_digest` table?

Reply:
We would have to do it both ways I guess?

Reply:
I agree it may be redundant. As a user, you shouldn't manage two different timezones. I should have been more explicit about it during the DR process, I felt it's a bit off

Reply:
Do you want to do both things in this PR? Removing timezone would be a bigger change to handle everywhere.

Reply:
Your call :)

Reply:
I'd like to solve timezones down the line.